pkg/plugins/builtin/httputil/httputil.go

```diff
@@ -110,3 +113,42 @@ func defaultHTTPClientFactory(cfg HTTPClientConfig) HTTPClient {
 	}
 	return c
 }
+
+func GetTransport(opts ...func(*http.Transport)) http.RoundTripper {
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+	for _, o := range opts {
+		o(tr)
+	}
+
+	_ = http2.ConfigureTransport(tr)
+	return tr
+}
+
+func WithTLSInsecureSkipVerify(v bool) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		t.TLSClientConfig.InsecureSkipVerify = v
+	}
+}
+
+func WithTLSClientAuth(cert tls.Certificate) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		t.TLSClientConfig.Certificates = []tls.Certificate{cert}
```

Comment:
Append instead to preserve existing certs ? Could that be necessary ? Guess not but the override tickled me, so just commenting for awarness.

Reply:
In this feature, this won't be needed, as only one client certificate can be provided.
As the `WithTLSClientAuth` could be used in the future, I will perform this change in case we need it.